dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/guestbook/GuestbookPage.java

```diff
@@ -238,54 +240,50 @@ public String save() {
             }
         }
 
-        Command<Dataverse> cmd;
-        try {
-            if (editMode == EditMode.CREATE || editMode == EditMode.CLONE) {
-                guestbook.setCreateTime(new Timestamp(new Date().getTime()));
-                guestbook.setUsageCount(new Long(0));
-                guestbook.setEnabled(true);
-                dataverse.getGuestbooks().add(guestbook);
-                cmd = new UpdateDataverseCommand(dataverse, null, null, dvRequestService.getDataverseRequest(), null);
-                commandEngine.submit(cmd);
-                create = true;
-            } else {
-                cmd = new UpdateDataverseGuestbookCommand(dataverse, guestbook, dvRequestService.getDataverseRequest());
-                commandEngine.submit(cmd);
-            }
+        if (editMode == EditMode.CREATE || editMode == EditMode.CLONE) {
+            Try.of(() -> guestbookService.saveGuestbook(guestbook))
+                    .onSuccess(dv -> JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("guestbook.create")))
+                    .onFailure(this::handleErrorMessages);
+        } else {
+            Try.of(() -> guestbookService.editGuestbook(guestbook))
+                    .onSuccess(dv -> JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("guestbook.save")))
+                    .onFailure(this::handleErrorMessages);
+        }
+
+        editMode = null;
+
+        return "/manage-guestbooks.xhtml?dataverseId=" + dataverse.getId() + "&faces-redirect=true";
+    }
+
+    public void cancel() {
+        editMode = null;
+    }
 
-        } catch (EJBException ex) {
+    // -------------------- PRIVATE --------------------
+    private void handleErrorMessages(Throwable throwable) {
+        if(throwable instanceof EJBException) {
             StringBuilder error = new StringBuilder();
-            error.append(ex).append(" ");
-            error.append(ex.getMessage()).append(" ");
-            Throwable cause = ex;
+            error.append(throwable).append(" ");
+            error.append(throwable.getMessage()).append(" ");
+            Throwable cause = throwable;
             while (cause.getCause() != null) {
                 cause = cause.getCause();
                 error.append(cause).append(" ");
                 error.append(cause.getMessage()).append(" ");
             }
-            //
-            FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle("guestbook.save.fail"), " - " + error.toString()));
+
+            FacesContext.getCurrentInstance().addMessage(null,
+                    new FacesMessage(FacesMessage.SEVERITY_FATAL, BundleUtil.getStringFromBundle("guestbook.save.fail"), " - " + error.toString()));
             logger.info("Guestbook Page EJB Exception. Dataverse: " + dataverse.getName());
```

Comment:
IMO this kinda logging is unnecessary, I think they just didn't knew how to log stacktrace so they are appending to string. This would be enough IMO `logger.log(Level.FINE, "Simple message", exception);`.

Reply:
This was not made for logging but for banner message. And it rather looks like error message is being specifically build for the banner. We might drop error details from banner and only log them.